lib/wallet/integration/src/Test/Integration/Scenario/CLI/Shelley/Transactions.hs

```diff
@@ -362,64 +362,48 @@ spec = describe "SHELLEY_CLI_TRANSACTIONS" $ do
                 ]
 
     it "TRANSMETA_CREATE_01b - \
-        \Transaction with metadata via CLI with simple metadata" $
-        \ctx -> runResourceT $ do
-
-        wSrc <- fixtureWallet ctx
-        wDest <- emptyWallet ctx
-        let amt = 10_000_000
-        let md = Just "{ \"1\": { \"string\": \"hello\" } }"
-        let expected = Just $
-                detailedMetadata $
-                TxMetadata $
-                    Map.singleton 1 (TxMetaText "hello")
-
-        args <- postTxArgs ctx wSrc wDest amt md Nothing
-        Stdout feeOut <- postTransactionFeeViaCLI ctx args
-        ApiFee (Quantity feeMin) (Quantity feeMax) _ _ <-
-            expectValidJSON Proxy feeOut
-
-        txJson <- postTxViaCLI ctx wSrc wDest amt md Nothing
-        verify txJson
-            [ expectCliField
-                (#amount . #getQuantity)
-                (between (feeMin + amt, feeMax + amt))
-            , expectCliField (#direction . #getApiT) (`shouldBe` Outgoing)
-            , expectCliField (#status . #getApiT) (`shouldBe` Pending)
-            , expectCliField #metadata (`shouldBe` expected)
-            ]
-
-        let wSrcId = T.unpack (wSrc ^. walletId)
-        let txId = getTxId txJson
+        \Transaction with metadata via CLI with simple metadata" $ \ctx ->
+        runResourceT $ do
+
+        -- Prepare test fixture:
```

Comment:
The thing is: this code is bad at telling the story (at a high level) so I resorted to comments. My goal was to enable readers (like me) to understand the high level test structure at a glance. Before I was struggling to understand it  as the test is overloaded with the low level details which make quite a noise. 

> Do you think this and the other comments could be dropped? If so it would avoid having the code and the comments compete about telling the story

not without substantially rewriting the code, such that it it uses abstractions (functions) to hide all the irrelevant details. I think that in the current situation comments represent an adequate trade-off between the effort spent and high-level structure understandability.